lib/extensions/extensions.js

```diff
@@ -225,6 +228,70 @@ function expandToIncludeShadowNunjucks (arr) {
   return out
 }
 
+function getCurrentPlugins () {
+  const pkg = fse.readJsonSync(pkgPath)
+  return Object.keys(pkg.dependencies).filter((dependency) => fse.pathExistsSync(pathToPackageConfigFile(dependency)))
+}
+
+let previousPlugins = getCurrentPlugins()
+
+async function waitForFile (filename, timeout, numberOfTestsPassed = 0) {
+  const delay = 100
+  const numberOfTestsPassedRequiredToConfirmReady = 3
+  return new Promise((resolve) => {
+    if (timeout < 0) {
+      resolve()
+    } else {
+      setTimeout(() => {
+        const stat = fs.statSync(filename)
+        if (stat.mtime < Date.now() - (numberOfTestsPassed + 1) * delay) {
+          numberOfTestsPassed++
+        } else {
+          numberOfTestsPassed = 0
+        }
+        if (numberOfTestsPassed < numberOfTestsPassedRequiredToConfirmReady) {
+          return waitForFile(filename, timeout - delay, numberOfTestsPassed)
+            .then(() => resolve())
+        } else {
+          resolve()
+        }
+      }, delay)
+    }
+  })
+}
```

Comment:
Could we achieve the same effect by using the Chokidar `awaitWriteFinish` option?

https://github.com/paulmillr/chokidar

Reply:
Replaced with the option as suggested :-)